### ISSUE TITLE
OOM crash when defining a custom xp drop rate for mobs from spawners

### ISSUE THREAD

**Issue by masmc05** [Archy-X/AureliumSkills#152]
### Server software and version

Purpur 1.19.2

### Expected behavior

The plugin would use a proper storage to see if the entity is from spawners (like weak hashmap), or have a special support for paper's api, or try to use persistet data containers when possible 

### Actual behavior

The plugin uses entity meta to get later if the mob is from spawners, because of that it's overloaded because this data remains even after a mob death or unload, and that state isn't saved after a server restart, so it becomes useless after restarts
![image](https://user-images.githubusercontent.com/63639746/206859916-e6da0493-237c-4049-98cf-1180892284d8.png)
so there's where it saves that state
https://github.com/Archy-X/AureliumSkills/blob/6ad95e51b1a3f4b234d3651a6b4acbdd5f510a80/src/main/java/com/archyx/aureliumskills/skills/fighting/FightingLeveler.java#L112

### How to reproduce

1 configure these values to be from 0 (inclusive) to 1 (exclusive)
![image](https://user-images.githubusercontent.com/63639746/206859923-4913540f-05b7-409d-bea1-2bce4e92f7bd.png)
2 have a 150 player server where they have lots of spawners

### Agreements

- [X] I am running the latest version of Aurelium Skills
- [X] I am not using an unsupported server software (https://wiki.aurelium.dev/skills/other/incompatibilities)
- [X] My version of Minecraft is supported by Aurelium Skills

**Comment by mastermc05** [comment]
and as this plugin now supports 1.13+, it can use https://hub.spigotmc.org/javadocs/spigot/org/bukkit/persistence/PersistentDataContainer.html to avoid this

**Comment by Archy-X** [comment]
I'm assuming PersistentDataContainer clears the memory when the mob dies? I'm fine with dropping 1.13 support to add this btw.

**Comment by mastermc05** [comment]
> I'm assuming PersistentDataContainer clears the memory when the mob dies? I'm fine with dropping 1.13 support to add this btw.

persistent data container saves the data in player's and mob's nbt, as soon as that nbt isn't saved anywhere, it's lost

**Comment by mastermc05** [comment]
but because players are always saved, their persistent data container will be saved as their inventory etc
and mob's data won't be saved on die, and will be unloaded from memory with the mob, so perfect for us

**Comment by mastermc05** [comment]
TL;DR
pdc is just like inventory in how it's stored, saved etc

**Comment by Archy-X** [comment]
Fixed in Beta 1.3.13
